src/main/java/vendingmachine/domain/Stock.java

```diff
@@ -0,0 +1,84 @@
+package vendingmachine.domain;
+
+import static vendingmachine.exception.ErrorCode.INVALID_PRODUCT_INFO;
+import static vendingmachine.exception.ErrorCode.INVALID_PRODUCT_REQUEST;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import vendingmachine.converter.Parser;
+import vendingmachine.dto.ProductDTO;
+
+/**
+ * 일급컬렉션
+ */
+public class Stock {
+    private final Map<String, Product> stockInfo;
+
+    private Stock(List<ProductDTO> productDTO) {
+        validate(productDTO);
+        this.stockInfo = Parser.parseToStockMap(productDTO);
+    }
+
+    public static Stock create(List<ProductDTO> stockInfo) {
+        return new Stock(stockInfo);
+    }
+
+    public int getMinGoodsPrice() {
+        return stockInfo.values()
+                .stream()
+                .filter(Product::hasStock)
+                .min(Product.priceComparator())
+                .map(Product::getPrice)
+                .orElse(0); // TODO: 0을 반환하면 안될듯
+    }
```

Comment:
상품을 가격으로 비교하는 Comparator를 따로 생성해주지 않고 스트림만으로도 기능을 구현할 수 있을 것 같아요!
또한 상품이 `Product`, `Goods` 두가지 용어가 사용되고 있어 통일해주면 더욱 좋을 것 같습니다!! 👍

```suggestion
    public int getMinProdutsPrice() {
        return stockInfo.values()
                .stream()
                .filter(Product::hasStock)
                .mapToInt(Product::getPrice)
                .min()
                .orElse(0);
    }
```

Reply:
> 상품을 가격으로 비교하는 Comparator를 따로 생성해주지 않고 스트림만으로도 기능을 구현할 수 있을 것 같아요! 또한 상품이 `Product`, `Goods` 두가지 용어가 사용되고 있어 통일해주면 더욱 좋을 것 같습니다!! 👍

아니 이럴수가 이런 방법이 있었네요..!! 좋은 의견 감사합니다! 👍